src/bin/scripts/vacuumdb.c
=======================================================
lhs: 100644 | ca6d00368325f6b9a27c21ecf00d9e022f9f4cb0
rhs: 100644 | 7e72db1edf7f287f8c8cf1c1a8c931c0bafc29a8
---@@ -393,9 +393,9 @@ vacuum_one_database(const char *dbname, vacuumingOptions *vacopts,
 		for (i = 0; i < ntups; i++)
 		{
 			appendPQExpBufferStr(&buf,
-							  fmtQualifiedId(PQserverVersion(conn),
-											 PQgetvalue(res, i, 1),
-											 PQgetvalue(res, i, 0)));
+								 fmtQualifiedId(PQserverVersion(conn),
+												PQgetvalue(res, i, 1),
+												PQgetvalue(res, i, 0)));
 
 			simple_string_list_append(&dbtables, buf.data);
 			resetPQExpBuffer(&buf);
@@ -412,6 +412,7 @@ vacuum_one_database(const char *dbname, vacuumingOptions *vacopts,
 			concurrentCons = ntups;
 		if (concurrentCons <= 1)
 			parallel = false;
+		PQclear(res);
 	}
 
 	/*

---